tests/effects/tdiagnostic_messages.nim

```diff
@@ -0,0 +1,32 @@
+discard """
+  errormsg: "\'a\' can have side effects"
+  nimout: '''  tdiagnostic_messages.nim(32, 30) Hint: 'a' calls 'callWithSideEffects' which has side effects
+    tdiagnostic_messages.nim(24, 6) Hint: 'callWithSideEffects' called by a `noSideEffect` routine, but has side effects
+      tdiagnostic_messages.nim(26, 34) Hint: 'callWithSideEffects' calls 'indirectCallViaVarParam' which has side effects
+        tdiagnostic_messages.nim(20, 6) Hint: 'indirectCallViaVarParam' called by a `noSideEffect` routine, but has side effects
+          tdiagnostic_messages.nim(21, 7) Hint: 'indirectCallViaVarParam' calls a routine with side effects via hidden pointer indirection
+      tdiagnostic_messages.nim(27, 33) Hint: 'callWithSideEffects' calls 'indirectCallViaPointer' which has side effects
+        tdiagnostic_messages.nim(22, 6) Hint: 'indirectCallViaPointer' called by a `noSideEffect` routine, but has side effects
+          tdiagnostic_messages.nim(23, 32) Hint: 'indirectCallViaPointer' calls a routine with side effects via pointer indirection
+      tdiagnostic_messages.nim(28, 8) Hint: 'callWithSideEffects' calls 'echo' which has side effects
+        ../../lib/system.nim(2004, 6) Hint: 'echo' called by a `noSideEffect` routine, but has side effects
```

Comment:
CI fails on windows because of this:
```
2021-07-04T17:29:51.2842395Z -        ../../lib/system.nim(2004, 6) Hint: 'echo' called by a `noSideEffect` routine, but has side effects
2021-07-04T17:29:51.2844010Z +        ..\..\lib\system.nim(2004, 6) Hint: 'echo' called by a `noSideEffect` routine, but has side effects
```


this could be fixed by `--filenames:canonical`, but then you'd have a 2nd problem: the test would become flaky as changes in system would break this test.

instead simply add a local proc in this file:
```nim
proc myEcho(a: string) {.sideEffect.} = discard
```